front/app/events/openSignUpInModal.ts

```diff
@@ -31,29 +15,59 @@ export interface ISignUpInMetaData {
   flow: TSignUpInFlow;
   pathname: string;
   verification?: boolean;
-  verificationContext?: ContextShape;
+  context?: AuthenticationContext;
   error?: ISignUpInError;
   isInvitation?: boolean;
   token?: string;
   noAutofocus?: boolean;
-  action?: () => void;
+  onSuccess?: () => void;
 }
 
-function emitOpenSignUpInModal(metaData: ISignUpInMetaData | undefined) {
-  eventEmitter.emit('openSignUpInModal', metaData);
+const OLD_MODAL_EVENT = 'openOldSignUpInModal';
+
+// Shared flow
+export async function openSignUpInModal(metaData?: Partial<ISignUpInMetaData>) {
+  if (metaData?.context) {
+    const response = await getAuthenticationRequirements(metaData.context);
+    const passwordNotRequired =
+      response.data.attributes.requirements.requirements.special.password ===
```

Comment:
This access (`esponse.data.attributes.requirements.requirements.special.password`) is super long - wonder if there is a utility function we could write to make this look cleaner?

Reply:
This is only a temporary patch to allow both flows to exist next to each other... will come up with something nicer!